src/aws.js

```diff
@@ -1,101 +1,132 @@
+source pre-runner-script.sh
+export RUNNER_ALLOW_RUNASROOT=1
+
+for idx in {1..${config.input.runnersPerMachine}}; do
+    cp -r ${config.input.runnerHomeDir} actions-runner-$idx
+    cd actions-runner-$idx
+    ./config.sh --url https://github.com/${config.githubContext.owner}/${config.githubContext.repo} --token ${githubRegistrationToken} --labels ${label} --name "$(hostname)-runner-$idx"
+    ./run.sh &
+    cd ..
+done
+`;
   } else {
-    return [
-      '#!/bin/bash',
-      'mkdir actions-runner && cd actions-runner',
-      `echo "${config.input.preRunnerScript}" > pre-runner-script.sh`,
-      'source pre-runner-script.sh',
-      'case $(uname -m) in aarch64) ARCH="arm64" ;; amd64|x86_64) ARCH="x64" ;; esac && export RUNNER_ARCH=${ARCH}',
-      'curl -O -L https://github.com/actions/runner/releases/download/v2.313.0/actions-runner-linux-${RUNNER_ARCH}-2.313.0.tar.gz',
-      'tar xzf ./actions-runner-linux-${RUNNER_ARCH}-2.313.0.tar.gz',
-      'export RUNNER_ALLOW_RUNASROOT=1',
-      `./config.sh --url https://github.com/${config.githubContext.owner}/${config.githubContext.repo} --token ${githubRegistrationToken} --labels ${label}`,
-      './run.sh',
-    ];
+    return `#!/bin/bash
+mkdir actions-runner && cd actions-runner
+echo "${config.input.preRunnerScript}" > pre-runner-script.sh
+source pre-runner-script.sh
+case $(uname -m) in aarch64) ARCH="arm64" ;; amd64|x86_64) ARCH="x64" ;; esac && export RUNNER_ARCH=\${ARCH}
+curl -O -L https://github.com/actions/runner/releases/download/v2.313.0/actions-runner-linux-\${RUNNER_ARCH}-2.313.0.tar.gz
+tar xzf ./actions-runner-linux-\${RUNNER_ARCH}-2.313.0.tar.gz
+cd ..
+export RUNNER_ALLOW_RUNASROOT=1
+
+for idx in {1..${config.input.runnersPerMachine}}; do
+    cp -r actions-runner actions-runner-$idx
+    cd actions-runner-$idx
+    ./config.sh --url https://github.com/${config.githubContext.owner}/${config.githubContext.repo} --token ${githubRegistrationToken} --labels ${label} --name "$(hostname)-runner-$idx"
+    ./run.sh &
+    cd ..
+done
+`;
+
   }
 }
 
+function buildMarketOptions() {
+  if (config.input.marketType !== 'spot') {
+    return undefined;
+  }
+
+  return {
+    MarketType: config.input.marketType,
+    SpotOptions: {
+      SpotInstanceType: 'one-time',
+    },
+  };
+}
+
 async function startEc2Instance(label, githubRegistrationToken) {
-  const ec2 = new AWS.EC2();
+  const ec2 = new EC2Client();
 
   const userData = buildUserDataScript(githubRegistrationToken, label);
 
   const params = {
     ImageId: config.input.ec2ImageId,
     InstanceType: config.input.ec2InstanceType,
-    MinCount: 1,
     MaxCount: 1,
-    UserData: Buffer.from(userData.join('\n')).toString('base64'),
+    UserData: Buffer.from(userData).toString('base64'),
     SubnetId: config.input.subnetId,
     SecurityGroupIds: [config.input.securityGroupId],
+    SubnetId: config.input.subnetId,
+    UserData: Buffer.from(userData.join('\n')).toString('base64'),
```

Comment:
UserData is duplicated
```suggestion
    UserData: Buffer.from(userData).toString('base64'),
    SubnetId: config.input.subnetId,
    SecurityGroupIds: [config.input.securityGroupId],
    SubnetId: config.input.subnetId,
```